x-pack/legacy/plugins/index_management/public/components/mappings_editor/components/document_fields/fields/field_types/text_type.tsx

```diff
@@ -0,0 +1,305 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React from 'react';
+import { EuiSpacer, EuiRange, EuiDualRange, EuiFormRow, EuiCallOut } from '@elastic/eui';
+
+import { NormalizedField, Field as FieldType } from '../../../../types';
+import {
+  UseField,
+  UseMultiFields,
+  Field,
+  FieldHook,
+  FormDataProvider,
+} from '../../../../shared_imports';
+import { getFieldConfig } from '../../../../lib';
+import { PARAMETERS_OPTIONS } from '../../../../constants';
+import { SelectWithCustom } from '../../../form';
+import {
+  StoreParameter,
+  IndexParameter,
+  BoostParameter,
+  EagerGlobalOrdinalsParameter,
+  NormsParameter,
+  SimilarityParameter,
+} from '../../field_parameters';
+import { EditFieldSection, EditFieldFormRow, AdvancedSettingsWrapper } from '../edit_field';
+
+interface Props {
+  field: NormalizedField;
+}
+
+const getDefaultValueToggle = (param: string, field: FieldType) => {
+  switch (param) {
+    case 'boost':
+    case 'position_increment_gap':
+    case 'similarity':
+    case 'term_vector': {
+      return field[param] !== undefined && field[param] !== getFieldConfig(param).defaultValue;
+    }
+    case 'analyzers': {
+      return field.search_analyzer !== undefined && field.search_analyzer !== field.analyzer;
+    }
+    case 'indexPrefixes': {
+      return (
+        field.index_prefixes !== undefined &&
```

Comment:
I find this boolean logic a bit difficult to parse (not impossible though) otherwise it would be nice to have typings running through here if possible.

Reply:
Good point, will make it clearer, thanks for raising it 👍